File: external/CppLogiMake/examples/fib/fib.py
#!/usr/bin/env python3
"""Python build of the shared fib demo: prints the sum of the first ten
Fibonacci numbers (== 88), matching the C++23, C++17 and Rust siblings so
the Fib.CrossLanguageEndToEnd test can assert cross-language agreement."""


def main() -> None:
    a, b, total = 0, 1, 0
    for _ in range(10):
        total += a
        a, b = b, a + b
    print(total)


if __name__ == "__main__":
    main()

File: external/CppLogiMake/examples/fib/fib.rs
// Rust build of the shared fib demo: prints the sum of the first ten
// Fibonacci numbers (== 88), matching the C++23, C++17 and Python
// siblings. The Fib.CrossLanguageEndToEnd test compiles this with
// `rustc fib.rs` when a Rust toolchain is present, and skips it
// (reporting SKIP) when rustc is not on PATH.
fn main() {
    let (mut a, mut b, mut total): (i64, i64, i64) = (0, 1, 0);
    for _ in 0..10 {
        total += a;
        let next = a + b;
        a = b;
        b = next;
    }
    println!("{}", total);
}
